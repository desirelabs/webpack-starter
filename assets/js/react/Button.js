import React from 'react'

class Buttons extends React.Component {
  constructor () {
    super()
    this.state = {
      text: ''
    }

    this.setData = this.setData.bind(this)
  }

  componentWillMount () {
    // this.setData()
  }

  setData () {
    this.setState({
      text: this.props.text
    })
  }

  actionClick () {
    this.setState({
      text: 'hello'
    })
  }

  render () {
    return (
      <div>
        <button onClick={() => this.actionClick()}>{this.state.text}</button>
      </div>
    )
  }
}

export default Buttons
