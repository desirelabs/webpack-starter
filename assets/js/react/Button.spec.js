import React from 'react'
import { configure, mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import renderer from 'react-test-renderer'

import Button from './Button'

configure({adapter: new Adapter()})

describe('Button component', () => {
  test('Button should button display', () => {
    const wrapper = renderer.create(<Button />).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  test('Should have prop text', () => {
    const action = jest.fn()
    const wrapper = shallow(<Button text='text' actionClick={action} />)
    expect(wrapper.instance().props.text.length).toBeGreaterThan(0)
  })

  /**
   * Test both that click works and state is updated
   */
  test('Click should trigger actionClick callback', () => {
    const mock = jest.fn()
    const wrapper = mount(<Button text='text' onActionClick={mock} />)
    wrapper.find('button').simulate('click')
    expect(wrapper.instance().state.text).toBe('hello')
  })
})
