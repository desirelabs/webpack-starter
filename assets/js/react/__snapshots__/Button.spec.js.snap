// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button should button display 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    
  </button>
</div>
`;
